File: querydsl-jpa/src/test/java/com/querydsl/jpa/QPerson.java
package com.querydsl.jpa;

import static com.querydsl.core.types.PathMetadataFactory.*;

import com.querydsl.core.types.path.*;

import com.querydsl.core.types.PathMetadata;
import javax.annotation.Generated;
import com.querydsl.core.types.Path;


/**
 * QPerson is a Querydsl querydsl type for Person
 */
@Generated("com.querydsl.codegen.EntitySerializer")
public class QPerson extends EntityPathBase<Person> {

    private static final long serialVersionUID = -219463259L;

    public static final QPerson person = new QPerson("person");

    public final StringPath firstName = createString("firstName");

    public final StringPath lastName = createString("lastName");

    public QPerson(String variable) {
        super(Person.class, forVariable(variable));
    }

    public QPerson(Path<? extends Person> path) {
        super(path.getType(), path.getMetadata());
    }

    public QPerson(PathMetadata<?> metadata) {
        super(Person.class, metadata);
    }

}

File: querydsl-scala/src/test/scala/com/querydsl/scala/QPerson.scala
package com.querydsl.scala

import com.querydsl.core.types.PathMetadataFactory._
import com.querydsl.core.types._

object QPerson extends QPerson("person"){
  override def as(variable: String) = new QPerson(variable)
  
}

class QPerson(cl: Class[_ <: Person], md: PathMetadata) extends EntityPathImpl[Person](cl, md) {
  def this(variable: String) = this(classOf[Person], forVariable(variable))

  def this(parent: Path[_], variable: String) = this(classOf[Person], forProperty(parent, variable))

  lazy val other = new QPerson(this, "other")

  val array = createArray[Array[String]]("array")

  val firstName = createString("firstName")

  val javaCollection = createCollection[String,StringPath]("javaCollection")

  val javaDouble = createNumber[java.lang.Double]("javaDouble")

  val javaInt = createNumber[Integer]("javaInt")
  
  val javaList = createList[String,StringPath]("javaList")

  val javaMap = createMap[String,String,StringPath]("javaMap")

  val javaSet = createSet[String,StringPath]("javaSet")

  val lastName = createString("lastName")

  val listOfPersons = createList[Person,QPerson]("listOfPersons")

  val scalaInt = createNumber[Int]("scalaInt")
  
  val scalaDouble = createNumber[Double]("scalaDouble")
  
  //val num = createNumber[String]("x")

  val scalaList = createList[String,StringPath]("scalaList")

  val scalaMap = createMap[String,String,StringPath]("scalaMap")

}